main.go

```diff
@@ -1,51 +1,166 @@
 package main
 
 import (
+	"bufio"
 	"encoding/json"
 	"fmt"
-	"io/ioutil"
+	"io"
 	"os"
 
 	"github.com/yazgazan/jaydiff/diff"
 )
 
 const (
-	statusUsage          = 2
-	statusReadError      = 3
-	statusUnmarshalError = 4
-	statusDiffError      = 5
-	statusDiffMismatch   = 6
+	statusUsage           = 2
+	statusReadError       = 3
+	statusUnexpectedError = 4
+	statusDiffError       = 5
+	statusDiffMismatch    = 6
 )
 
 var (
 	// Version is replaced by the tag when creating a new release
 	Version = "dev"
 )
+type nextJson func() (interface{}, int, error)
 
 func main() {
-	var err error
+	var (
+		lhs_file,rhs_file *os.File
+		err error
+	)
+
 	conf := readConfig()
 
-	lhs := parseFile(conf.Files.LHS)
-	rhs := parseFile(conf.Files.RHS)
+	// Open files for reading
+	lhs_file, err = os.Open(conf.Files.LHS)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error: cannot read %s: %s\n", conf.Files.LHS, err.Error())
+		os.Exit(statusReadError)
+	}
+	defer lhs_file.Close()
+	rhs_file, err = os.Open(conf.Files.RHS)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error: cannot read %s: %s\n", conf.Files.RHS, err.Error())
+		os.Exit(statusReadError)
+	}
+	defer rhs_file.Close()
+
+	//
+	if conf.JsonLines {
+		os.Exit(compareLoop(getNextJsonByLine(lhs_file), getNextJsonByLine(rhs_file),&conf))
+	} else {
+		os.Exit(compareLoop(getNextJson(lhs_file), getNextJson(rhs_file),&conf))
+	}
+}
+
+func compareLoop(lhs_next, rhs_next nextJson, conf *config) int {
+	var is_eof bool
+	var lhs_err,rhs_err, err error
+	var lhs,rhs interface{}
+	var lhs_cnt, rhs_cnt, rc, rc_final int
+
+	rc_final = 0
+	is_eof = false
+
+	for !is_eof {
+		lhs, lhs_cnt, lhs_err = lhs_next()
+		rhs, rhs_cnt,rhs_err = rhs_next()
+
+		// If both file reach EOF
+		switch {
+		case lhs_err == io.EOF && rhs_err == io.EOF:
+			is_eof = true
+			continue
+		case lhs_err != nil && lhs_err != io.EOF:
+			fmt.Fprintln(os.Stderr, lhs_err.Error())
+			return statusUnexpectedError
```

Comment:
Having one file returning `io.EOF` while the other still has values is not unexpected when running a diff on JSON streams. These should be treated as missing/excess values in the diff, resulting in a `ContentDiffer`, and the appropriate pretty printing.